analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -267,6 +267,72 @@ public void Branching_LearnsObjectConstraint_Binary_Negated_NullableBool_CS(stri
+        {
+            var validator = CreateIfElseEndValidatorCS("(bool)arg", OperationKind.Conversion, "bool?");
+            validator.ValidateTag("If", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.True));
+            validator.ValidateTag("Else", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.False));
+            validator.TagValues("End").Should().SatisfyRespectively(
+                x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.True),
+                x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.False));
+        }
+
+        [TestMethod]
+        public void Branching_LearnsObjectConstraint_Integer_VB()
+        {
+            var validator = CreateIfElseEndValidatorVB("arg", OperationKind.Binary, "Integer");
+            validator.ValidateTag("If", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull));
+            validator.ValidateTag("Else", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull));
+            validator.TagValues("End").Should().SatisfyRespectively(x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull));
+        }
+
+        [TestMethod]
+        public void Branching_LearnsObjectConstraint_NullableInteger_VB()
+        {
+            var validator = CreateIfElseEndValidatorVB("arg", OperationKind.Binary, "Integer?");
+            validator.ValidateTag("If", x => x.Should().HaveNoConstraints("arg.GetValueOrDefault() is called in 'if arg' and arg is not found in RoslynSymbolicExecution.SetBranchingConstraints"));
+            validator.ValidateTag("Else", x => x.Should().HaveNoConstraints());
+            validator.TagValues("End").Should().SatisfyRespectively(x => x.Should().HaveNoConstraints());
+        }
+
+        [DataTestMethod]
+        [DataRow("True", true)]
+        [DataRow("False", false)]
+        public void Branching_LearnsObjectConstraint_NullableBool_VB(string branchValue, bool expectedPath)
+        {
+            var validator = SETestContext.CreateVB($$"""
+                Dim b As Boolean? = {{branchValue}}
+                if b Then
+                    Tag("True", b)
+                else
+                    Tag("False", b)
+                End if
+                Tag("End", b)
+                """).Validator;
+            validator.ValidateTagOrder(branchValue, "End");
+            validator.ValidateTag(branchValue, x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expectedPath)));
+            validator.TagValues("End").Should().SatisfyRespectively(x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, BoolConstraint.From(expectedPath)));
+        }
+
+        [TestMethod]
+        public void Branching_LearnsObjectConstraint_NullableBool_UnknownVB()
+        {
+            var validator = SETestContext.CreateVB($$"""
+                Dim b As Boolean? = Unknown(Of Boolean?)
```

Comment:
There's no difference between `Arg As Boolean?` and `Unknown(Of Boolean?)`. So this UT should be made short using the `CreateIfElseEndValidatorVB` like in the first new UTs above